test/e2e2/conftest.py

```diff
@@ -238,17 +251,101 @@ def query(self, qname: str, qtype: str):
         try:
             section = dns.message.AUTHORITY if qtype == dns.rdatatype.from_text('NS') else dns.message.ANSWER
             response = answer.find_rrset(section, qname, dns.rdataclass.IN, qtype)
-            print(f'DNS <<< {response}')
+            tsprint(f'DNS <<< {response}')
             return {i.to_text() for i in response.items}
         except KeyError:
-            print('DNS <<< !!! not found !!! Complete Answer below:\n' + answer.to_text())
+            tsprint('DNS <<< !!! not found !!! Complete Answer below:\n' + answer.to_text())
             return {}
 
 
 class NSLordClient(NSClient):
     where = os.environ["DESECSTACK_IPV4_REAR_PREFIX16"] + '.0.129'
 
 
+class Replication:
+
+    def query(self, zone: str, qname: str, qtype: str, covers: str = None):
+        if qtype == 'RRSIG':
+            assert covers, 'If querying RRSIG, covers parameter must be set to a RR type, e.g. SOA.'
+        else:
+            assert not covers
+            covers = dns.rdatatype.NONE
+
+        zonefile = os.path.join('/zones', zone + '.zone')
+        zone = dns.name.from_text(zone, origin=dns.name.root)
+        qname = dns.name.from_text(qname, origin=zone)
+
+        assert os.path.exists(zonefile), \
+            f'While checking that {qname}/{qtype} was correctly written into the replication, the zone file ' \
+            f'could not be found at {zonefile}. Number of zones in /zones: ' \
+            f'{len(list(filter(lambda f: f.endswith(".zone"), os.listdir("/zones"))))}.'
+
+        try:
+            tsprint(f'RPL >>> {qname}/{qtype} in {zone}')
+            z = dns.zone.from_file(f=zonefile, origin=zone, relativize=False)
+            v = {i.to_text() for i in z.find_rrset(qname, qtype, covers=covers).items}
+            tsprint(f'RPL <<< {v}')
+            return v
+        except KeyError:
+            tsprint(f'RPL <<< RR Set {qname}/{qtype} not found')
+            return {}
+        except dns.zone.NoSOA:
+            tsprint(f'RPL <<< Zone {zone} not found')
+            return None
+
+
+@pytest.fixture()
+def replication() -> Replication:
+    return Replication()
```

Comment:
Why does this need to be a fixture? It just serves as a library here, doesn't it?

Reply:
fixed